zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -73,21 +72,24 @@
  */
 public class Paragraph extends Job implements Cloneable, JsonSerializable {
 
-  private static final long serialVersionUID = -6328572073497992016L;
-
   private static Logger logger = LoggerFactory.getLogger(Paragraph.class);
-  private transient InterpreterFactory factory;
-  private transient InterpreterSettingManager interpreterSettingManager;
+  private static Pattern REPL_PATTERN = Pattern.compile("%([\\w\\.]+).*", Pattern.DOTALL);
```

Comment:
use Reg expression to parse text which is much more concise.

Reply:
hmm,... I think we break this often when we change it, could we have some more tests for this?
for example, does this work properly if the paragraph text is 
```
foo %3
```

Reply:
Add more test in `ParagraphTest.java`